atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTaskTest.java

```diff
@@ -0,0 +1,433 @@
+                        context.startTimestampInclusive() - 1,
+                        context.endTimestampExclusive()))
+                .thenReturn(SweepBatchWithPartitionInfo.of(
+                        SweepBatch.of(
+                                ImmutableSet.of(),
+                                DedicatedRows.of(ImmutableList.of()),
+                                context.endTimestampExclusive() - 1),
+                        ImmutableSet.of()));
+        when(boundRetriever.getStrictUpperBoundForCompletelyClosedBuckets()).thenReturn(context.bucketIdentifier());
+
+        defaultSingleBucketSweepTask.runOneIteration(context.sweepableBucket());
+        verify(sweepQueueDeleter).sweep(eq(ImmutableList.of()), eq(Sweeper.of(context.shardAndStrategy())));
+        verify(sweepQueueCleaner)
+                .clean(
+                        eq(context.shardAndStrategy()),
+                        eq(ImmutableSet.of()),
+                        eq(context.endTimestampExclusive() - 1L),
+                        eq(DedicatedRows.of(ImmutableList.of())));
+        verify(bucketProgressStore)
+                .updateBucketProgressToAtLeast(
+                        context.bucket(), context.completeProgressForBucket().orElseThrow());
+        verifyNoInteractions(completionListener);
+    }
+
+    @ParameterizedTest
+    @MethodSource("allSweepBuckets")
+    public void sweepsEmptyBucketUpToSweepTimestamp(SweepBucketTestContext context) {
+        long sweepTimestamp = context.startTimestampInclusive() + MIN_BUCKET_SIZE - 1;
+
+        when(sweepTimestampSupplier.getAsLong()).thenReturn(sweepTimestamp);
+        when(sweepQueueReader.getNextBatchToSweep(
+                        context.shardAndStrategy(), context.startTimestampInclusive() - 1, sweepTimestamp))
+                .thenReturn(SweepBatchWithPartitionInfo.of(
+                        SweepBatch.of(ImmutableSet.of(), DedicatedRows.of(ImmutableList.of()), sweepTimestamp - 1),
+                        ImmutableSet.of()));
+
+        defaultSingleBucketSweepTask.runOneIteration(context.sweepableBucket());
+        verify(sweepQueueDeleter).sweep(eq(ImmutableList.of()), eq(Sweeper.of(context.shardAndStrategy())));
+        verify(sweepQueueCleaner)
+                .clean(
+                        eq(context.shardAndStrategy()),
+                        eq(ImmutableSet.of()),
+                        eq(sweepTimestamp - 1),
+                        eq(DedicatedRows.of(ImmutableList.of())));
+        verify(bucketProgressStore)
+                .updateBucketProgressToAtLeast(
+                        context.bucket(),
+                        BucketProgress.createForTimestampProgress(
+                                sweepTimestamp - 1 - context.startTimestampInclusive()));
+        verifyNoInteractions(completionListener);
+    }
+
+    @ParameterizedTest
+    @MethodSource("allSweepBuckets")
+    public void doesNotRemoveSweepDataOutsideBucket(SweepBucketTestContext context) {
```

Comment:
note to self to check tomorrow: Is this just checking that we don't also remove the fine partition right before startTimestampInclusive?

Reply:
Yes, this test was added after I noticed that you need to handle that case correctly!